packages/suite/src/services/coingecko.ts

```diff
@@ -1,12 +1,56 @@
+/* eslint-disable max-classes-per-file */
 import { LastWeekRates, TickerId } from '@wallet-types/fiatRates';
 import FIAT_CONFIG from '@suite-config/fiat';
+import { differenceInMilliseconds } from 'date-fns';
 
+// const COINGECKO_API_BASE_URL = 'https://api.coingecko.com/api/v3';
 const COINGECKO_API_BASE_URL = 'https://cdn.trezor.io/dynamic/coingecko/api/v3';
 
 interface HistoricalResponse extends LastWeekRates {
     symbol: string;
 }
 
+class RateLimiter {
+    // Poor man's coingecko rate limiter
+    // Slows down requests so there is `delayMs` gap between them
+    public delayMs: number; // gap between each request
+    public lastFetchTimestamp: number;
+    private queued: number; // how many requests are waiting to be fired
+    private totalDelay: number; // by how much time are we gonna slown down next request
+
+    constructor(delayMs: number) {
+        this.delayMs = delayMs;
+
+        this.lastFetchTimestamp = 0;
+        this.queued = 0;
+        this.totalDelay = 0;
+    }
+
+    async limit(fn: any) {
+        const msSinceLastFetch = differenceInMilliseconds(
+            new Date().getTime(),
+            this.lastFetchTimestamp,
+        );
+        if (msSinceLastFetch < this.delayMs) {
+            this.queued += 1;
+            this.totalDelay += this.delayMs;
+            // dummy wait for this.totalDelay before we fire next request
+            await new Promise(resolve => setTimeout(resolve, this.totalDelay)); // slow down firing next request
+        }
+
+        this.lastFetchTimestamp = new Date().getTime();
+        const results = await fn();
+        this.queued -= 1;
+        if (this.queued === 0) {
+            // if all queued requests were fired, we need to reset totalDelay to properly delay next batch of requests
+            this.totalDelay = 0;
+        }
+        return results;
+    }
+}
+
+const limiter = new RateLimiter(1000);
```

Comment:
nit:
```suggestion
const rateLimiter = new RateLimiter(1000);
```

I would change it because you use use it below like this `limiter.limit(() => fetch(url))` and it is kinda painful for me 😆

Reply:
jeez ok :D